oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/time/ISODateTimeParser.java

```diff
@@ -40,6 +43,16 @@
   public static ZonedDateTime parseISODateTime(String isoDateTimeString) throws Exception {
     ZonedDateTime zdt = null;
 
+    isoDateTimeString = isoDateTimeString.trim();
+
+    if (isoDateTimeString.startsWith("-")) {
+      throw new Exception("Negative Dates are not supported: " + isoDateTimeString);
```

Comment:
can we maybe get rid of the generic `Exception` class here? I know this is quite a bit off-topic from the initial pull request goal (to fix #259), but adding new cases of such "bad code" is not great either.

Reply:
Hey @tyrasd , can you explain why this is "bad code"? And do we have a guideline or should write one which defines such cases as "bad code" with suggestions for "good code".

Reply:
well, maybe "bad code" was not the best wording. But sonarlint classifies this as _code smell_ with _major severity_: https://rules.sonarsource.com/java/RSPEC-112

> Using such generic exceptions as Error, RuntimeException, Throwable, and Exception prevents calling methods from handling true, system-generated exceptions differently than application-generated errors. See also http://cwe.mitre.org/data/definitions/397.html and https://wiki.sei.cmu.edu/confluence/display/java/ERR07-J.+Do+not+throw+RuntimeException%2C+Exception%2C+or+Throwable

Reply:
What kind of suggestion should we do here.
We could use a already existing exception like:

1. java.time.format.DateTimeParseException

Reply:
IMO option 3 (defining our own exception like ISODateTimeParserException) or 2 would be better in this instance because this here is a specific exception of the choices we made in the OSHDB here. But option 1 would still be better than the generic `Exception` class.

Reply:
Hi, I leave it up to you to choose a "well smelling" Exception flavor. That's not my expertise. So I would be glad if someone else takes over the exception handling. Functionally the issue is fixed so far and the class is much safer to use as before and well tested. So feel free to assign someone else for the exceptions.

Reply:
thanks, I will take care of the rest :+1: